### PR TITLE
geo/geomfn: add a microbenchmark for st_intersects

### PR DESCRIPTION
**Pull request by andyyang890**
This patch adds a microbenchmark for st_intersects
so that we can quantify any performance improvement
when we implement caching.

Release note: None

### CODE REVIEW

**Comment by andyyang890** [comment]
Should I commit the benchmark results with each commit too?

e.g. this file
```
BenchmarkIntersects-16    	 1000000	      1237 ns/op
BenchmarkIntersects-16    	  963583	      1118 ns/op
BenchmarkIntersects-16    	 1000000	      1164 ns/op
BenchmarkIntersects-16    	 1000000	      1092 ns/op
BenchmarkIntersects-16    	 1000000	      1114 ns/op
BenchmarkIntersects-16    	 1000000	      1191 ns/op
BenchmarkIntersects-16    	 1000000	      1152 ns/op
BenchmarkIntersects-16    	 1000000	      1278 ns/op
BenchmarkIntersects-16    	 1000000	      1214 ns/op
BenchmarkIntersects-16    	 1000000	      1145 ns/op
```

File: pkg/geo/geomfn/binary_predicates_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+var benchRes bool
+
+func BenchmarkIntersects(b *testing.B) {
+	var res bool
+	var err error
+	for n := 0; n < b.N; n++ {
+		for _, polygon := range benchPolygons {
+			for _, point := range benchPoints {
+				res, err = Intersects(polygon, point)
```
Comment (otan) [inline pkg/geo/geomfn/binary_predicates_bench_test.go:54]: we should wrap this line around `b.StartTimer()` and `b.StopTimer()`

File: pkg/geo/geomfn/binary_predicates_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		geo.MustParseGeometry("POINT (1.22081126074562 6.133882930268385)"),
+		geo.MustParseGeometry("POINT (-85.860041139453415 11.249996764058892)"),
+		geo.MustParseGeometry("POINT (27.449993447588497 -7.300033753593539)"),
+		geo.MustParseGeometry("POINT (-73.340004314859073 -37.480000387828738)"),
+	}
+)
+
+var benchRes bool
+
+func BenchmarkIntersects(b *testing.B) {
```
Comment (otan) [inline pkg/geo/geomfn/binary_predicates_bench_test.go:48]: let's rename this `BenchmarkPointAndLargePolygonIntersect`

i wouldn't mind adding another mini benchmark for smaller polygons / points, so we have at least two data sets that are interesting!

File: pkg/geo/geomfn/binary_predicates_bench_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	var res bool
+	var err error
```
Comment (otan) [inline pkg/geo/geomfn/binary_predicates_bench_test.go:50]: let's call b.StopTimer() and b.ResetTimer() before the first iteration

File: pkg/geo/geomfn/binary_predicates_bench_test.go (hunk 1 above)
Comment (andyyang890) [reply pkg/geo/geomfn/binary_predicates_bench_test.go:54]: I just gave this a try but it seems to cause the benchmarking to not terminate. My guess would be it somehow prevents the results from converging.

**Review by otan — APPROVED** [review]

**Comment by andyyang890** [comment]
Hmm something weird I noticed is that the benchmark result for the small one is a lot larger, which seems kinda strange:
```
name                                 time/op
IntersectsLargePolygonsAndPoints-16  1.01µs ±10%
IntersectsSmallPolygonsAndPoints-16  15.2µs ± 9%
```

**Comment by andyyang890** [comment]
bors r=otan

**Comment by andyyang890** [comment]
bors r=otan

**Comment by andyyang890** [comment]
bors r=otan

**Comment by andyyang890** [comment]
bors r=otan

**Pull request merged**